### ISSUE TITLE
Seems to slow down youtube considerably

### ISSUE THREAD

**Issue by Nodrance** [ajayyy/SponsorBlock#2093]
When I have sponsorblock enabled, the player lags considerably. Playback isn't affected, but seeking definitely is. It can have a delay of 15-20 seconds after i seek before it does it. 

I suspect this is a cross compatability issue, here is my list of currently enabled extensions
![image](https://github.com/user-attachments/assets/8f8ddf25-4cc2-4495-a8ed-b156456d53ff)
![image](https://github.com/user-attachments/assets/a53d4cb4-76a9-4a46-88e4-2143a102a0d0)
![image](https://github.com/user-attachments/assets/10311159-07e4-42d9-aa32-aa55df567ce1)

This has persisted across several months, with updates to both firefox and sponsorblock in that time

**Comment by X13X** [comment]
Yeah something is going on behind the scenes that noone is talking about.

I have a baremetal system that seems to lag whenever SB is enabled, and yet noone wants to talk about this -_-

**Comment by AvianAnalyst** [comment]
I'm also having this issue. Pausing and playing hte video also has a lag of many seconds before it happens. Also if I close a tab playing a youtube video, I continue to get audio from the video for several seconds after the tab is closed. If skipping is disable, but the extension is still enabled, I don't have these issues. If i enable it and refresh the page it causes the same problems immediately

**Comment by yowidin** [comment]
Same here. Checking `about:processes` shows that most of the time is spent in GC 🤔

**Comment by sskodje** [comment]
I just had to uninstall the extension for this reason. When installed and browsing through a channels video feed, the extension runs code and network requests for every video scrolled through, causing quite large delays and huge memory usage + lag due to garbage collection. I can't see why the extension should need to run any code at all for just displaying a thumbnail you haven't even clicked on.
